### PR TITLE
ddl: move modify-column job to add-index job queue

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: https://github.com/pingcap/tidb/issues/24275

Problem Summary: Since backfill column-type-change is quite similar to add an index, it may take a long time to read and write data. Thus, to prevent blocking other "instant" DDL jobs, column-type-change jobs should be put to the adding-index job queue.

### What is changed and how it works?

What's Changed: Move backfill column-type-change job to the adding-index queue.

### Check List 

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

File: ddl/ddl_worker.go [hunk 1]
```diff
@@ -319,6 +305,24 @@ func (d *ddl) addBatchDDLJobs(tasks []*limitJobTask) {
...
+func resolveJobListKey(tp model.ActionType) meta.JobListKeyType {
+	if tp == model.ActionAddIndex ||
+		tp == model.ActionAddPrimaryKey ||
+		tp == model.ActionModifyColumn {
+		return meta.AddIndexJobListKey
+	}
+	return meta.DefaultJobListKey
+}
+
+func resolveJobListKeyNeg(tp model.ActionType) meta.JobListKeyType {
```
Comment (wjhuang2016) [inline ddl/ddl_worker.go:317]: `resolveJobListKeyNeg` is a bad name. I don't know what it means till I read the code. `resolveOtherJobListKey` maybe a better name?

File: ddl/ddl_worker_test.go [hunk 2]
```diff
@@ -1574,35 +1574,37 @@ func (s *testDDLSuite) TestParallelDDL(c *C) {
...
+				finishedJobs = finishedJobs[len(finishedJobs)-12:]
 				// check some jobs are ordered because of the dependence.
 				c.Assert(finishedJobs[0].ID, Equals, job1.ID)
 				c.Assert(finishedJobs[1].ID, Equals, job2.ID)
 				c.Assert(finishedJobs[2].ID, Equals, job3.ID)
 				c.Assert(finishedJobs[4].ID, Equals, job5.ID)
-				c.Assert(finishedJobs[10].ID, Equals, job11.ID)
+				c.Assert(finishedJobs[11].ID, Equals, job12.ID)
 				// check the jobs are ordered in the adding-index-job queue or general-job queue.
 				addIdxJobID := int64(0)
```
Comment (zimulala) [inline ddl/ddl_worker_test.go:1603]: Update `addIdxJobID`？

**Review by zimulala — APPROVED** [review]
LGTM

**Pull request merged**